### PR TITLE
[script] [carve-lockpicks] Create carve-lockpicks.lic

### PR DESCRIPTION
**Pull request by Raykyn55**
Lockpick carving script for thieves.

### CODE REVIEW

File: carve-lockpicks.lic [hunk 1]
```diff
@@ -0,0 +1,114 @@
+=begin
+	Need these settings in your yaml:
+		#Stow locations for Grandmaster's, Master's, and anything less.
+    grand_container: backpack
```
Comment (rpherbig) [inline carve-lockpicks.lic:4]: Please add any new settings to `base.yaml`

File: carve-lockpicks.lic [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+  def initialize
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @belt = @settings.engineering_belt
+	  @grandbag = @settings.grand_container
+	  @masterbag = @settings.master_container
+	  @trashbag = @settings.trash_container
+	  @initial = @settings.initial
+    @tool = "carving knife"
```
Comment (rpherbig) [inline carve-lockpicks.lic:23]: Why use an instance var when this is a constant string?

File: carve-lockpicks.lic [hunk 3]
```diff
@@ -0,0 +1,114 @@
+=begin
+	Need these settings in your yaml:
+		#Stow locations for Grandmaster's, Master's, and anything less.
+    grand_container: backpack
+    master_container: haversack
+    trash_container: cloak
+    #Whether to initial Grandmaster's with an extra carve.
+    initial: true
```
Comment (rpherbig) [inline carve-lockpicks.lic:8]: A lot of these setting names are kind of generic. A few ideas:

* Rename them to indicate they're for carving lockpicks
* Nest them under a parent setting which indicates they're for carving lockpicks
* Something else?

File: carve-lockpicks.lic [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    @bag = @settings.crafting_container
+    @belt = @settings.engineering_belt
+	  @grandbag = @settings.grand_container
+	  @masterbag = @settings.master_container
+	  @trashbag = @settings.trash_container
+	  @initial = @settings.initial
+    @tool = "carving knife"
+    DRC.wait_for_script_to_complete('buff', ['carve'])
+    DRC.bput('sit', 'You sit', 'You are already sitting', 'You rise', 'While swimming?') unless sitting?
+    DRCC.get_crafting_item("carving knife", @crafting_container, @crafting_container_items, @belt)
```
Comment (rpherbig) [inline carve-lockpicks.lic:26]: If you do keep `@tool`, should it be used here?

File: carve-lockpicks.lic [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      stow_item("master")
+    when /but feel your knife slip and the iron keyblank snaps like a twig/
+      get_keyblank
+    end
+  end
+
+  def carve_lockpick
+    case DRC.bput('carve my lockpick with my knife', /proudly glance down at a grandmaster/, /proudly glance down at a master/, /into the side of a/, /find a creature to carve/, /It would be better/, /Working with extreme caution/, /but feel your knife slip/)
+    when /proudly glance down at a grandmaster/
+      initial_grand if @initial
```
Comment (rpherbig) [inline carve-lockpicks.lic:71]: What do you think about moving the conditional into a guard clause in `initial_grand`?

File: carve-lockpicks.lic [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    when /proudly glance down at a master/
+      stow_item("master")
+    when /but feel your knife slip/
+      get_keyblank
+	  return
+    when /find a creature to carve/, /It would be better/
+      stow_item("trash")
+    when /Working with extreme caution/
+      carve_lockpick
+    when /iron lockpick in your hands./
```
Comment (rpherbig) [inline carve-lockpicks.lic:82]: Is the `.` needed in the regex?

File: carve-lockpicks.lic (hunk 5 above)
Comment (Raykyn55) [reply carve-lockpicks.lic:72]: I think I understand what you mean. Updated, it's now under stow_item, where it carves again IF @initial_grand = true

File: carve-lockpicks.lic (hunk 3 above)
Comment (Raykyn55) [reply carve-lockpicks.lic]: nested under lockpick_carve_settings

File: carve-lockpicks.lic (hunk 2 above)
Comment (Raykyn55) [reply carve-lockpicks.lic]: removed.

File: carve-lockpicks.lic (hunk 6 above)
Comment (Raykyn55) [reply carve-lockpicks.lic]: removed.

**Comment by Raykyn55** [comment]
Thank you for the comments and suggestions, still have a lot to learn!

File: carve-lockpicks.lic (hunk 5 above)
Comment (Raykyn55) [reply carve-lockpicks.lic:71]: Which removes that method, forgot that part. Second updated version is correct.

Comment (rpherbig) [reply carve-lockpicks.lic:71]: What I meant was a guard clause in the `initial_grand` function:

```
def initial_grand
  return unless @initial
  ...
end
```

But I'm fine with the current approach, as long as you remove line 71 (which calls the now removed function).

Comment (Raykyn55) [reply carve-lockpicks.lic:71]: It's being stubborn, but I think it's gone now.

**Review by rpherbig — APPROVED** [review]

**Comment by Dartellum** [comment]
I thought someone made one. Checking.... I have a make-lockpicks.lic script already so it exists. Perhaps it on the repo (not GitHub)?

**Pull request merged**